docs/ref/templates/api.txt

```diff
@@ -922,38 +922,189 @@ option. It uses each loader until a loader finds a match.
 
 .. _custom-template-loaders:
 
+.. currentmodule:: django.template.loaders.base
+
 Custom loaders
---------------
+==============
 
-Custom ``Loader`` classes should inherit from
-``django.template.loaders.base.Loader`` and override the
-``load_template_source()`` method, which takes a ``template_name`` argument,
-loads the template from disk (or elsewhere), and returns a tuple:
-``(template_string, template_origin)``.
+It's possible to load templates from additional sources using custom template
+loaders. Custom ``Loader`` classes should inherit from
+``django.template.loaders.base.Loader`` and define the ``get_contents()`` and
+``get_template_sources()`` methods.
 
 .. versionchanged:: 1.8
 
     ``django.template.loaders.base.Loader`` used to be defined at
     ``django.template.loader.BaseLoader``.
 
-The ``load_template()`` method of the ``Loader`` class retrieves the template
-string by calling ``load_template_source()``, instantiates a ``Template`` from
-the template source, and returns a tuple: ``(template, template_origin)``.
+.. versionchanged:: 1.9
 
-.. currentmodule:: django.template
+In previous versions of Django, custom loaders defined a single method:
+``load_template_source``.
+
+Loader methods
+--------------
+
+.. class:: Loader
+
+    Loads templates from a given source, such as the filesystem or
+    a database.
+
+    .. method:: get_template_sources(self, template_name, template_dirs=None)
+
+        A method that takes a ``template_name`` and yields ``LoaderOrigin``
+        instances for each possible source.
+
+        For example, the filesystem loader may receive ``index.html`` as a
+        ``template_name`` argument.  ``get_template_sources`` would yield
+        origins for the full path of ``index.html`` as it appears in each
+        template directory the loader looks at.
+
+        ``get_template_sources`` doesn't need to verify that the file exists
+        at a given path, but it should ensure the path is valid. For instance,
+        the filesystem loader makes sure the path lies under a valid template
+        directory.
+
+        .. warning::
+
+            The ``template_dirs`` argument has been deprecated. Loaders that
```

Comment:
Since you're introducing a new API and the deprecated argument is just there for backwards-compatibility within Django, I agree that you shouldn't document it.

Django already uses `getargspec` in a few places to account for the removal of arguments from documented APIs.